api/hypershift/v1beta1/azure_types.go

```diff
@@ -1,144 +1,177 @@
 	//
 	// If this field is not supplied, the Service Principal credentials will be written to a file on the disk of each VM
 	// in order to be accessible by the cloud provider; the aforementioned credentials provided are the same ones as
 	// HostedCluster.Spec.Platform.Azure.Credentials. However, this is less secure than using a managed identity.
 	//
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
+	//
 	// +optional
 	MachineIdentityID string `json:"machineIdentityID,omitempty"`
 }
 
-// AzureVMImage represents the different types of image data that can be provided for an Azure VM.
+// AzureVMImage represents the different types of boot image sources that can be provided for an Azure VM.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'ImageID' ?  has(self.imageID) : !has(self.imageID)",message="imageID is required when type is ImageID, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'AzureMarketplace' ?  has(self.azureMarketplace) : !has(self.azureMarketplace)",message="azureMarketplace is required when type is RequiredMember, and forbidden otherwise"
 // +union
 type AzureVMImage struct {
-	// Type is the type of image data that will be provided to the Azure VM. This can be either "ImageID" or
-	// "AzureMarketplace".
+	// type is the type of image data that will be provided to the Azure VM.
+	// Valid values are "ImageID" and "AzureMarketplace".
+	// ImageID means ... TODO
+	// AzureMarketplace means ... TODO
 	//
 	// +kubebuilder:validation:Required
-	// +kubebuilder:validation:Enum:=ImageID;AzureMarketplace
 	// +unionDiscriminator
 	Type AzureVMImageType `json:"azureImageType"`
 
-	// ImageID is the Azure resource ID of a VHD image to use to boot the Azure VMs from.
+	// imageID is the Azure resource ID of a VHD image to use to boot the Azure VMs from.
+	// TODO: What is the valid character set for this field? What about minimum and maximum lengths?
 	//
 	// +optional
 	// +unionMember
 	ImageID *string `json:"imageID,omitempty"`
 
-	// AzureMarketplace contains the Azure Marketplace image info to use to boot the Azure VMs from.
+	// azureMarketplace contains the Azure Marketplace image info to use to boot the Azure VMs from.
 	//
 	// +optional
 	// +unionMember
 	AzureMarketplace *MarketplaceImage `json:"azureMarketplace,omitempty"`
 }
 
-// MarketplaceImage specifies the information needed to create an Azure VM from an Azure Marketplace image. This struct
-// replicates the same fields found in CAPZ - https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go.
+// MarketplaceImage specifies the information needed to create an Azure VM from an Azure Marketplace image.
+// + This struct replicates the same fields found in CAPZ - https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go.
```

Comment:
I don't think this information is supposed to end up in the kubectl explain output is it? Prefixing with a `+` skips it from ending up there